### ISSUE TITLE
Firebase Analytics for iOS Background Fetch.

### ISSUE THREAD

**Issue by EzlyJohnnie** [firebase/firebase-ios-sdk#6161]
* Xcode version: 11.4 (11E146)
  * Firebase SDK version: 6.26.0
  * Firebase Component: Analytics
  * Component version: 6.4.1
  * Installation method: CocoaPods

I am trying to log event to Firebase Analytics when app is launched for background fetch.
Based on the [document](https://support.google.com/firebase/answer/9191807?hl=en), `session_start` should be only logged when the app is in the foreground. However, I found sessions are created during the background fetch. 

Is it a bug? Or please advise how should I avoid it.

**Comment by EzlyJohnnie** [comment]
Hi guys, any updates on this?

**Comment by premjubl** [comment]
Any updates on this?

**Comment by paulb777** [comment]
Thanks for the reports and sorry about the lack of response here. We're tracking the issue internally at b/196036854

**Comment by EzlyJohnnie** [comment]
Just need to enable [background fetch](https://developer.apple.com/documentation/uikit/app_and_environment/scenes/preparing_your_ui_to_run_in_the_background/updating_your_app_with_background_app_refresh) for iOS app.
When the background fetch process is kicked in by the system, `session_start` is logged (which I believe should not).

**Comment by allenktv** [comment]
We are not able to reproduce this on our end. Do you have debug logs that show this behavior?

**Comment by iOS-developer-9** [comment]
Hi, we are experiencing similar issue. 

Firebase SDK version: 8.7.0
Firebase Component: Analytics
Installation method: CocoaPods

Source what "wakes up" our app is still unknown (periodically for some increments of 30min x Y). Can't reproduce but we see in production from the logs that session_start is triggered from background for lots of users. But what we would expect that session_start is not reported in this events as per documentation.

Don't know if anyone else is experiencing same issue?

**Comment by romkoth** [comment]
> Hi, we are experiencing similar issue.
> 
> Firebase SDK version: 8.7.0 also on 8.9.1 (unfortunately did not helped: https://firebase.google.com/support/release-notes/ios) Firebase Component: Analytics Installation method: CocoaPods
> 
> Source what "wakes up" our app is still unknown (periodically for some increments of 30min x Y). Can't reproduce but we see in production from the logs that session_start is triggered from background for lots of users. But what we would expect that session_start is not reported in this events as per documentation.
> 
> For now we have noticed on iOS15.x.y. But is this coincidence we are not sure(on iOS14 is also happening but lot less occurrence, maybe also the user base is with iOS14 is also lot less).
> 
> Don't know if anyone else is experiencing same issue?

Yes we also had such issue, after Apple released iOS 15 number of session_start was increased in few times without any reason, after moving from 8.3.1 to 8.10.0 we see less amount of events, but still in a few times bigger than it should be, does anyone found the way to fix this?

**Comment by paulb777** [comment]
@iOS-developer-9 @romkoth Thanks for the reports. iOS 15 introduced pre-warming which can cause extra `session_start` events. We're investigating, but do not yet have a solution. We're tracking internally at  b/207324460.

Some additional context at https://twitter.com/steipete/status/1466013492180312068

**Comment by tsunghung** [comment]
This issue is caused by the life cycle inconsistency of Swift apps on iOS 15 : https://developer.apple.com/forums/thread/696539.

**Comment by tsunghung** [comment]
Sorry, my previous comment is not precise.
This issue is caused by the app life cycle inconsistency in UIScene based apps on iOS 15 when pre-warming.

**Comment by tsunghung** [comment]
I tried [another Swift app](https://github.com/firebase/quickstart-ios/tree/master/analytics/LegacyAnalyticsQuickstart/AnalyticsExampleSwift), which does not support scenes, and `application(_:didFinishLaunchingWithOptions:)` is not called during prewarming.
It seems that apps supports UIScene life-cycle would have different behaviors during prewarming.

**Comment by corey-plugshare** [comment]
We are experiencing this issue as well. Is there an update or ETA on a fix? Thanks!

**Comment by tsunghung** [comment]
We have not got any responds or solutions form Apple, so we implemented a workaround in Firebase/Analytics v8.12.0, which will be released in about 2 weeks.
